### PR TITLE
provide HelpKeyword information for offline help

### PR DESCRIPTION
**Pull request by heejaechang**

### CODE REVIEW

**Comment by heejaechang** [comment]
@mavasani can you take a look? this gives information for error list to show offline help viewer in offline mode.

File: src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.BuildTableDataSource.cs [hunk 1]
```diff
@@ -193,6 +193,9 @@ public override bool TryGetValue(int index, string columnName, out object conten
                         case StandardTableKeyNames.ErrorCodeToolTip:
                             content = GetHelpLinkToolTipText(data);
                             return content != null;
+                        case StandardTableKeyNames.HelpKeyword:
```
Comment (mavasani) [inline src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.BuildTableDataSource.cs]: @heejaechang Looking at the PR I realized I only added handler for SuppressionStateColumnDefinition.ColumnName in live source but not for build source, I will make that fix tomorrow.

**Comment by mavasani** [comment]
:+1:

**Comment by mavasani** [comment]
I think you also need to add this fix to FxCop: http://index/#StanCore/TableDataSource/CodeAnalysisTableEntriesSnapshot.cs,110?

File: src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.BuildTableDataSource.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.BuildTableDataSource.cs]: There is a lot of redundancy in this method.  Why not have a "GetValue" method that just returns things like ```data.id```.  Then have htis method be:

```C#
bool TryGetValue(...)
{
    var value = GetValue(...);
    return value != null;
}
```

It should cut down the size of this method (and the repeated code) by nearly half.

**Comment by heejaechang** [comment]
@mavasani good point. I will make the change in FxCop as well.

File: src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.BuildTableDataSource.cs (hunk 1 above)
Comment (heejaechang) [inline src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.BuildTableDataSource.cs]: @CyrusNajmabadi some of them could, but I dont think it is that bad to require cleaning up.

**Comment by heejaechang** [comment]
@dotnet-bot retest linux please

**Comment by heejaechang** [comment]
@dotnet-bot retest linux please

**Comment by heejaechang** [comment]
@dotnet-bot retest this please

**Comment by heejaechang** [comment]
@dotnet-bot retest this please

**Comment by heejaechang** [comment]
@dotnet-bot retest this please

**Comment by heejaechang** [comment]
@dotnet-bot retest this please

**Comment by heejaechang** [comment]
Jenkins seems in weird state. each tests (actually not "test" but test runner) are failing due to arbitrary reasons each time I re-run. and each failure looks like nothing to do with my change.

**Comment by jmarolf** [comment]
test prtest/lin/dbg/unit32 please
test prtest/mac/dbg/unit32 please

**Comment by heejaechang** [comment]
test prtest/win/dbg/eta please

**Comment by heejaechang** [comment]
test win eta please

**Comment by heejaechang** [comment]
@dotnet-bot retest win eta please

**Comment by heejaechang** [comment]
test win eta please

**Pull request merged**